django-stubs/template/backends/base.pyi

```
@@ -9,8 +12,15 @@ class BaseEngine:
     def __init__(self, params: Mapping[str, Any]) -> None: ...
     @property
     def app_dirname(self) -> Optional[str]: ...
-    def from_string(self, template_code: str) -> Template: ...
-    def get_template(self, template_name: str) -> Optional[Template]: ...
+    def from_string(self, template_code: str) -> EngineTemplate: ...
+    def get_template(self, template_name: str) -> Optional[EngineTemplate]: ...
```

Looks like this exception is not returned, but raised: https://github.com/django/django/blob/d80a2585532ef15184afed6cf4ec1358198a894d/django/template/loader.py#L15